libs/pika/async_mpi/include/pika/async_mpi/mpi_polling.hpp

```diff
@@ -61,17 +74,18 @@ namespace pika::mpi::experimental {
         /// This should be used with great caution as setting it too low can
         /// cause deadlocks. The default value is size_t(-1) - i.e. unlimited
         /// The value can be set using an environment variable as follows
-        /// PIKA_MPI_MSG_THROTTLE=512
+        /// PIKA_MPI_MSG_THROTTLE=64
         /// but user code setting it will override any default or env value
         /// This function returns the previous throttling threshold value
-        PIKA_EXPORT size_t set_max_requests_in_flight(size_t);
+        PIKA_EXPORT std::uint32_t set_max_requests_in_flight(
+            std::uint32_t, stream_index = stream_index(-1));
```

Comment:
What does the `-1` represent? Could we use an optional instead?

Reply:
the `-1` is the set `all` queues/streams with the same value. An optional is a good idea. I'll do that. (and add a comment to document why)